### ISSUE TITLE
Implement GroupNorm, LayerNorm & InstanceNorm

### ISSUE THREAD

**Issue by seanpmorgan** [tensorflow/addons#6]
Per the [RFC](https://github.com/tensorflow/community/blob/393dfc9e03dd4150a839de6357a26e0259aa97d6/rfcs/20181214-move-to-addons.md), we need to move [layer_norm](https://github.com/tensorflow/tensorflow/blob/v1.12.0/tensorflow/contrib/layers/python/layers/layers.py#L2198) from contrib to addons:

This will involve restructuring in an OO format by inheriting from base Keras Layer, modifying the code to match those APIs, and modifying test cases to run in TF2.

https://arxiv.org/abs/1607.06450

**EDIT:** It will make more sense to implement GroupNorm, and then created interfaces for LayerNorm and InstanceNorm

Group Norm: https://arxiv.org/pdf/1803.08494.pdf

**Comment by lgeiger** [comment]
What do you think about implementing this together with #7 using a [`GroupNormalization` Layer](https://github.com/keras-team/keras-contrib/blob/master/keras_contrib/layers/normalization.py#L415) which is a generalized case of layer normalization and instance normalization?

**Comment by seanpmorgan** [comment]
@lgeiger Thanks for the suggestion! This would be much more succinct. 

I would vote to have interfaces named LayerNorm and InstanceNorm which are GroupNorm with 1 group and input_dim# groups respectively.

**Comment by Smokrow** [comment]
I will give this a try
